data_structures/disjoint_set/disjoint_set.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * A Disjoint Set is a data structure that keeps track of a set of elements
+ * partitioned into a number of disjoint (non-overlapping) subsets.
+ * Elements are uniquely represented by an index (0-based).
+ * 
+ * It uses union by size: The smaller set is joined to the bigger one.
+ * This allows to keep the `find` complexity to O(logn).
+ * 
+ * You can perform the following operations on the disjoint set:
+ * - find: Determine which subset a particular element is in - O(logn) ~ O(1) with path compression.
```

Comment:
Since you are implementing path compression here, I'd remove the misleading note of O(log n)